File: Tutoriels/Python/Projets/src/manager/GSetting.py
#================================================
from manager.GConfig import GConfig
#================================================
class GSetting:
    #================================================
    m_instance = None
    #================================================
    G_CHAR_COMMENT = "#"
    #================================================
    def __init__(self):
        if GSetting.m_instance == None:
            GSetting.m_instance = self        
    #================================================
    @staticmethod
    def Instance():
        if GSetting.m_instance == None:
            GSetting()
        return GSetting.m_instance
    #================================================
    def load(self, file):
        lFile = open(file, "r")
        for lLine in lFile:
            lTrim = lLine.strip()
            if(not lTrim): continue
            if(lTrim[0] == self.G_CHAR_COMMENT): continue
            lSplit = lTrim.split("=")
            lKey = lSplit[0].strip()
            lValue = lSplit[1].strip()
            GConfig.Instance().setData(lKey, lValue)
        GConfig.Instance().showData()
#================================================

File: Tutoriels/CSharp/Projets/src/manager/GSetting.cs
//===============================================
using System;
using System.Collections.Generic;
using System.IO;
//===============================================
public class GSetting {
    //===============================================
    private static GSetting m_instance = null;
    private const char G_CHAR_COMMENT = '#';
    //===============================================
    private GSetting() {

    }
    //===============================================
    public static GSetting Instance() {
        if(m_instance == null) {
            m_instance = new GSetting();
        }
        return m_instance;
    }
    //===============================================
    public void load(string file) {
        string[] lLineMap = File.ReadAllLines(file);
        char[] lSepMap = new char[] {'='};
        foreach(string lLine in lLineMap) {
            string lTrim = lLine.Trim();
            if(lTrim == "") continue;
            if(lTrim[0] == G_CHAR_COMMENT) continue;
            string[] lSplit = lTrim.Split(lSepMap, StringSplitOptions.RemoveEmptyEntries);
            if(lSplit.Length < 2) continue;
            string lKey = lSplit[0].Trim();
            string lValue = lSplit[1].Trim();
            GConfig.Instance().SetData(lKey, lValue);
        }
        GConfig.Instance().ShowData();
    }
    //===============================================
}
//===============================================
